src/Core/Domain/Alias/Command/UpdateSearchTermAliasesCommand.php

```diff
@@ -28,60 +28,52 @@
 
 namespace PrestaShop\PrestaShop\Core\Domain\Alias\Command;
 
-use PrestaShop\PrestaShop\Core\Domain\Alias\ValueObject\AliasId;
+use PrestaShop\PrestaShop\Core\Domain\Alias\ValueObject\SearchTerm;
 
-/**
- * Edits alias with given data
- */
-class UpdateAliasCommand
+class UpdateSearchTermAliasesCommand
 {
-    private AliasId $aliasId;
+    private SearchTerm $oldSearchTerm;
+    private SearchTerm $newSearchTerm;
 
     /**
-     * @var string[]
-     */
-    private array $aliases;
-
-    private string $searchTerm;
-
-    /**
-     * @param int $aliasId
-     * @param string[] $aliases this input is array of string aliases that are returned from the form input
-     * @param string $searchTerm
+     * @param string $oldSearchTerm
+     * @param string $newSearchTerm
+     * @param array{
+     *   array{
+     *     alias: string,
+     *     active: bool,
+     *   }
+     * } $aliases
      */
     public function __construct(
-        int $aliasId,
-        array $aliases,
-        string $searchTerm,
+        string $oldSearchTerm,
+        string $newSearchTerm,
+        private array $aliases
```

Comment:
I wonder if new term should be optional?
```suggestion
        string $searchTerm,
        private array $aliases,
        string $newSearchTerm = null,
```
Meaning when `$newSearchTerm` is null it is simply an update of the aliases list, which should be the majority of the use cases, we could even check that both term are indeed different